src/main/java/com/khutircraftubackend/category/CategoryService.java

```diff
@@ -95,16 +98,19 @@ public void deleteCategory(Long id) {
 
     @Transactional
     public void deleteCategory(Long id, boolean forceDelete) {
-        // check if category with the given id exists.
-        // if not, them method below will throw an error:
-        findCategoryById(id);
+        List<ProductEntity> products = productRepository.findAllByCategory(findCategoryById(id));
```

Comment:
i guess we haven't implemented logic yet that checks for underlying products in a category that is requested to be deleted. So my suggestion to remove this line but create a Tech-Debt jira ticket which describes this logic.